src/parser.ts

```diff
@@ -13,11 +16,15 @@ enum TOKEN_TYPES {
   closeBlock = 'CLOSE_BLOCK',
   openBlock = 'OPEN_BLOCK',
   command = 'COMMAND',
+  include = 'INCLUDE',
 }
 
+type FluentBitToken = Token & { filePath: string };
+
 const stateSet = {
```

Comment:
I have no issue with the code itself but at some point we could do with visualising the state machine in comments or documentation. There's a few decent tools for it so I might try to do it as a PR so I understand it all. Code-based FSMs become unwieldy very quickly even for the author.

Reply:
yea, that sounds dandy. lemme know if you get around it.

Reply:
Resolved.